### PR TITLE
Refactor of NOONE_EXTESNIONS:

### PR DESCRIPTION
**Pull request by NoOneBlood**
All functions and definitions was moved to nnexts.h and nnexts.cpp files
to minimize ifdefs and generally make code more clear

### CODE REVIEW

File: source/blood/src/ai.cpp [hunk 1]
```diff
@@ -128,6 +116,20 @@ void aiNewState(spritetype *pSprite, XSPRITE *pXSprite, AISTATE *pAIState)
         pAIState->enterFunc(pSprite, pXSprite);
 }
 
+bool isUmmune(spritetype* pSprite, int dmgType, int minScale) {
```
Comment (CommonLoon102) [inline source/blood/src/ai.cpp]: Ummune?

File: source/blood/src/nnexts.h [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+extern short gPhysSpritesList[kMaxSuperXSprites];
+extern short gProxySpritesCount;
+extern short gSightSpritesCount;
+extern short gPhysSpritesCount;
+
+// - FUNCTIONS ------------------------------------------------------------------
+bool nnExtEraseModernStuff(spritetype* pSprite, XSPRITE* pXSprite);
+void nnExtInitModernStuff(bool bSaveLoad);
+void nnExtProcessSuperSprites(void);
+bool nnExtIsUmmune(spritetype* pSprite, int dmgType, int minScale = 16);
```
Comment (CommonLoon102) [inline source/blood/src/nnexts.h]: Should be Immune?

Comment (NoOneBlood) [reply source/blood/src/nnexts.h]: nnExtIsImmune and isImmune acts a bit differently, i left isImmune for Cerberus fix only, and nnExtIsImmune could normally work only ifdefs is enabled

Comment (CommonLoon102) [reply source/blood/src/nnexts.h]: I know, but you wrote isUmmune and nnExtIsUmmune instead isImmune and nnExtIsImmune, so just a typo I guess (`I` is next to `U` on the keyboard).

Comment (NoOneBlood) [reply source/blood/src/nnexts.h]: oh, thanks :D

**Pull request merged**